python/tank_vendor/shotgun_api3/lib/sgutils.py

```diff
@@ -0,0 +1,115 @@
+"""
+ -----------------------------------------------------------------------------
+ Copyright (c) 2009-2024, Shotgun Software Inc.
+
+ Redistribution and use in source and binary forms, with or without
+ modification, are permitted provided that the following conditions are met:
+
+  - Redistributions of source code must retain the above copyright notice, this
+    list of conditions and the following disclaimer.
+
+  - Redistributions in binary form must reproduce the above copyright notice,
+    this list of conditions and the following disclaimer in the documentation
+    and/or other materials provided with the distribution.
+
+  - Neither the name of the Shotgun Software Inc nor the names of its
+    contributors may be used to endorse or promote products derived from this
+    software without specific prior written permission.
+
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+"""
+
+# This module contains addtional functions and variables to supplement the six
+# module.
+
+import io
+import sys
+
+# For python 3, the `file` type no longer exists, and open() returns an
+# io.IOBase instance. We add file_types to allow comparison across python
+# versions.  See https://stackoverflow.com/questions/36321030#36321030
+#
+# This means that to test if a variable contains a file in Python3
+
+file_types = (io.IOBase, )
+
+# For python-api calls that result in an SSL error, the exception raised is
+# different on Python 3. Store the approriate exception class in a
+# variable to allow easier exception handling across Python 3.
+import ssl
+ShotgunSSLError = ssl.SSLError
+
+
+def normalize_platform(platform):
+    """
+    Normalize the return of sys.platform in Python 3.
+
+    This function will normalize platform strings to always conform to
+    Python3 behavior.
+
+    :param str platform: The platform string to normalize
+    :param bool python2: The python version behavior to target.  If True, a
+        Python2-style platform string will be returned (i.e. 'linux2'), otherwise
+        the modern 'linux' platform string will be returned.
+
+    :returns: The normalized platform string.
+    :rtype: str
+    """
+    return "linux" if platform.startswith("linux") else platform
+
+
+platform = normalize_platform(sys.platform)
```

Comment:
I think this attribute is only used in tk-core and not in other TK components. ([search](https://github.com/search?q=org%3Ashotgunsoftware+%22six.platform%22++NOT+is%3Aarchived&type=code&p=2))

In this case, I think we can get rid of it and replace all occurrences to use something native